.github/workflows/build.yml

```diff
@@ -1,19 +1,15 @@
-name: Deploy spec
+name: Build spec
 
-on: [push]
+on: pull_request
```

Comment:
this should happen on push too.

Reply:
The deploy job will already do a build, no?

Reply:
on main, but not on other branches pushed to the repo

Reply:
Do we _want- it to run for other branches pushed to the repo which are neither PRs nor main?

Personally I tend to make a bunch of commits on a branch while iterating, and sometimes I push those so I can pick up on a different machine, but I don't want the build job to run on all of those commits. They're not ready yet; that's why they're not in a PR.

Reply:
Sure, and in those cases you can just ignore any transient failures, but it's very useful in proposals to have long-running branches with potential in-flight changes, and having CI run on those is also useful.

Reply:
Usually you'd do those as a PR, I would think? Or, of course, edit this file on your branch to list your branch.

The default across pretty much every project I've worked on is that CI runs only on the main branch, possibly some other explicitly named branches, and open PRs. (For example, [node](https://github.com/nodejs/node/blob/360cda192653475588391c6694fac115bff72d56/.github/workflows/linters.yml#L3-L11), [prettier](https://github.com/prettier/prettier/blob/5b113e71b1808d6916f446c3aa49c3c53e3bdb98/.github/workflows/dev-test.yml#L3-L8), [eslint](https://github.com/eslint/eslint/blob/55534f1a7040fad94bb5726759fbb9acb60d1c24/.github/workflows/ci.yml#L2-L6), and [typescript](https://github.com/microsoft/TypeScript/blob/7c6521e8fb8f24a31f28e00e92838ad1b975fbb3/.github/workflows/ci.yml#L3-L11) all work this way.) For this template repository I would very much like to stick to that default unless there is some extremely strong reason to be special.

Reply:
As we've seen in the past, we've got very different sets of anecdata about projects we've worked on - most of the ones I've worked on run CI on every branch. It's mostly the very rare niche cases of large projects with multiple collaborators - like node, prettier, eslint, typescript, etc - that bother skipping CI on non-main branches, since on most projects, contributions are all made from forks.

Reply:
Well, it is also the case that people who have set up their own CI for proposals have followed my convention, not yours: [proposal-temporal](https://github.com/tc39/proposal-temporal/blob/7671bd28cc9b18ce943be78dc343cca14b1858c1/.github/workflows/lint.yml#L2), [proposal-class-static-block](https://github.com/tc39/proposal-class-static-block/blob/c0cabee0aa2d036a8d902fea7bc1d179e3de2477/.github/workflows/pr.yml#L2-L4), [proposal-change-array-by-copy](https://github.com/tc39/proposal-change-array-by-copy/blob/main/.github/workflows/test.yml#L2-L5), etc.